src/batch/src/executor/row_seq_scan.rs

```diff
@@ -443,7 +469,7 @@ impl<S: StateStore> RowSeqScanExecutor<S> {
                     },
                 ),
                 ordered,
-                PrefetchOptions::new_for_large_range_scan(),
+                PrefetchOptions::new_with_exhaust_iter(limit.is_none()),
```

Comment:
Is it possible to pass the specific `limit` to the iterator?

Reply:
There is another related PR https://github.com/risingwavelabs/risingwave/pull/13558 which provide a new prefetch strategy. I think we can decide to use which prefetch strategy based on the `limit` later. Passing the specific limit to the iterator which I am not sure how much it could benefit. cc @hzxa21